patches/webrtc/streaming_crash_fixes.patch

```diff
@@ -0,0 +1,49 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Alex Ciarlillo <alex.ciarlillo@gmail.com>
+Date: Thu, 19 Aug 2021 18:45:49 -0400
+Subject: streaming crash fixes
+
+
+diff --git a/modules/video_coding/codecs/h264/h264_encoder_impl.cc b/modules/video_coding/codecs/h264/h264_encoder_impl.cc
+index 66861e6e742714bb5d1d8bc78a8934a4edb71b77..d70531d9e9a4bc06ec22e8b7750ef4bb866bdcec 100644
+--- a/modules/video_coding/codecs/h264/h264_encoder_impl.cc
++++ b/modules/video_coding/codecs/h264/h264_encoder_impl.cc
+@@ -255,10 +255,7 @@ int32_t H264EncoderImpl::InitEncode(const VideoCodec* inst,
+ 
+     // Create downscaled image buffers.
+     if (i > 0) {
+-      downscaled_buffers_[i - 1] = I420Buffer::Create(
+-          configurations_[i].width, configurations_[i].height,
+-          configurations_[i].width, configurations_[i].width / 2,
+-          configurations_[i].width / 2);
++      downscaled_buffers_[i - 1] = I420Buffer::Create(configurations_[i].width, configurations_[i].height);
```

Comment:
why do we change the constructor here if we bail early below?

Reply:
These are 2 separate fixes. One if for when a window is closed while streaming. This one occurs when streaming a window of an odd with. Specifically I streamed a window that was 289pix wide. This resulted in a stride_u of 144. 

It then fails the check in here:
```
I420Buffer::I420Buffer(int width,
                       int height,
                       int stride_y,
                       int stride_u,
                       int stride_v)
    : width_(width),
      height_(height),
      stride_y_(stride_y),
      stride_u_(stride_u),
      stride_v_(stride_v),
      data_(static_cast<uint8_t*>(
          AlignedMalloc(I420DataSize(height, stride_y, stride_u, stride_v),
                        kBufferAlignment))) {
  RTC_DCHECK_GT(width, 0);
  RTC_DCHECK_GT(height, 0);
  RTC_DCHECK_GE(stride_y, width);
  RTC_DCHECK_GE(stride_u, (width + 1) / 2);
  RTC_DCHECK_GE(stride_v, (width + 1) / 2);
}
```

Because it calculates the expected stride differently than how it is passed from `h264_encoder_impl` and is expecting `145`.

The alternate constructor just does the right thing by default:

```
I420Buffer::I420Buffer(int width, int height)
    : I420Buffer(width, height, width, (width + 1) / 2, (width + 1) / 2) {}
```